cucumber-junit-platform-engine/src/main/java/io/cucumber/junit/platform/engine/CucumberTestEngine.java

```diff
@@ -39,11 +42,24 @@ public String getId() {
 
     @Override
     public TestDescriptor discover(EngineDiscoveryRequest discoveryRequest, UniqueId uniqueId) {
-        CucumberEngineDescriptor engineDescriptor = new CucumberEngineDescriptor(uniqueId);
+        TestSource testSource = createEngineTestSource(discoveryRequest);
+        CucumberEngineDescriptor engineDescriptor = new CucumberEngineDescriptor(uniqueId, testSource);
         new DiscoverySelectorResolver().resolveSelectors(discoveryRequest, engineDescriptor);
         return engineDescriptor;
     }
 
+    private static TestSource createEngineTestSource(EngineDiscoveryRequest discoveryRequest) {
+        // Workaround. Test Engines do not normally have test source.
```

Comment:
If there is an issue logged with Maven, it might be good to mention (the link to) that here?